static/app/components/sidebar/sidebarAccordion.tsx

```diff
@@ -78,6 +81,65 @@ function SidebarAccordion({children, ...itemProps}: SidebarAccordionProps) {
 
 export {SidebarAccordion};
 
+function recursivelyFindChildren(
+  children: React.ReactNode,
+  componentName: string,
+  found: Array<React.ReactNode> = []
+): React.ReactNode {
+  Children.toArray(children).forEach(child => {
+    if (!isAReactElement(child)) {
+      return;
+    }
+
+    if (child && child?.props && child?.props?.children) {
+      recursivelyFindChildren(child.props.children, componentName, found);
+      return;
+    }
+
+    if (typeof child.type === 'string') {
+      if (child.type === componentName) {
+        found.push(child);
+      }
+    } else {
+      if (child?.type?.name === componentName) {
+        found.push(child);
+      }
+    }
+
+    return;
+  });
+
+  return found;
+}
+
+function isAReactElement(element: React.ReactNode): element is React.ReactElement {
```

Comment:
Can you use https://react.dev/reference/react/isValidElement?

Reply:
Honestly embarrassing I didn't see this function being used _in this file_. Yes! Thank you! Changed